src/main/java/de/tum/in/www1/artemis/service/CourseService.java

```diff
@@ -110,6 +123,21 @@ public Course findOneWithExercisesAndLecturesForUser(Long courseId, User user) {
         return course;
     }
 
+    private void filterLectureUnits(Course course) {
+        for (Lecture lecture : course.getLectures()) {
+            if (lecture.getLectureUnits() == null) {
+                return;
+            }
+
+            List<LectureUnit> visibleLectureUnits = lecture.getLectureUnits().stream().filter(Objects::nonNull).filter(LectureUnit::isVisibleToStudents)
+                    .collect(Collectors.toList());
+            // do not combine into two streams as order of execution is not determined and can lead to null pointer
```

Comment:
I don't understand that. How does this cause a NPE? What could not be deterministic? Unless there are weird side effects, I don't see how that could happen.

Reply:
I changed it and added explaining comment in 9b3402385e9a1bd07006d0d1b7d84f66f2ec8c3d

Reply:
Makes sense now, thanks.